liveblog.php

```diff
@@ -777,13 +882,20 @@ private static function add_default_plupload_settings() {
 	 * @return string
 	 */
 	private static function get_entries_endpoint_url() {
-		$post_permalink = get_permalink( self::$post_id );
-		if ( false !== strpos( $post_permalink, '?p=' ) )
-			$url = add_query_arg( self::url_endpoint, '', $post_permalink ) . '='; // returns something like ?p=1&liveblog=
-		else
-			$url = trailingslashit( trailingslashit( $post_permalink ) . self::url_endpoint ); // returns something like /2012/01/01/post/liveblog/
-		$url = apply_filters( 'liveblog_endpoint_url', $url, self::$post_id );
-		return $url;
+		if (self::use_rest_api && self::can_use_rest_api()) {
+			$url = trailingslashit( trailingslashit( WPCOM_Liveblog_Rest_Api::$endpoint_base ) . self::$post_id );
+			$url = apply_filters( 'liveblog_endpoint_url', $url, self::$post_id );
+			return $url;
+		} else {
+			$post_permalink = get_permalink( self::$post_id );
+			if ( false !== strpos( $post_permalink, '?p=' ) ) {
+				$url = add_query_arg( self::url_endpoint, '', $post_permalink ) . '='; // returns something like ?p=1&liveblog=
+			} else {
+				$url = trailingslashit( trailingslashit( $post_permalink ) . self::url_endpoint ); // returns something like /2012/01/01/post/liveblog/
+			}
+			$url = apply_filters( 'liveblog_endpoint_url', $url, self::$post_id );
+			return $url;
```

Comment:
Instead of repeating the filter and return twice, we can just move it outside the `if/else`.